### DISCUSSION TITLE
Having trouble getting ghost cms api working in sveltekit with typescript eneabled

### DISCUSSION THREAD

**Discussion by NafunaAfrica** [sveltejs/kit#9538] [Q&A] [answered]
Hi there.

So I have a sveltekit app running my site. I have a ghost cms installed on a separate server and would like to get the content into my sveltekit. I am having trouble as the code 
`import  GhostContentAPI  from '@tryghost/content-api';` throws up red error lines under the @tryghost. The blog doesn't show up.

I am using this code to test:
```svelte
<script context='module' lang="ts">
  
import { GhostContentAPI } from '@tryghost/content-api';


const GHOST_URL = 'http://blog.nafuna.tv';
const GHOST_KEY = '87f6096411ae42f96df2615620';
const GHOST_VERSION = 'v5';

export async function load() {
  const api = GhostContentAPI({
      url: GHOST_URL,
      key: GHOST_KEY,
      version: GHOST_VERSION})
  const postsJson = await api.posts.browse({limit: 5, include: 'tags,authors'});

  return {
    props: {
      postsJson: postsJson
    }
  }
}


</script>

<script lang="ts">
    import type { PageData } from './$types';
    export let postsJson; 
    console.log(postsJson.title);
    export let data: PageData;
</script>


<div class="hero bg-base-200 p-5">
    <div class="hero-content text-center">
      <div class="max-w-md">
        <h1 class="text-5xl font-bold">Coming Soon</h1>
        <p class="py-6">We're working on this section and it will be live in the next few weeks!</p>
        <a href="/agency" class="btn btn-primary">Back to Agency</a>
      </div>
    </div>
  </div>

  {#each postsJson as post}
  <h1> {post.title} </h1>
  <h3> {post.excerpt} </h3>
{/each}
```

not sure what I am missing. A lot of the code I got from the ghost docs

**Comment by teemingc** [comment, marked as answer]
Hi Nafuna,

The code example from the ghost docs is using an outdated convention of loading data from Svelte's beta days.
Declaring the load function in the svelte file is no longer supported.

You'll have to utilise a `+page.server.ts` file to load your data into a `+page.svelte` page file as explained in the link below.
https://kit.svelte.dev/docs/load

The solution should look something like the two file examples below:
```ts
// src/routes/+page.server.ts
import { GhostContentAPI } from '@tryghost/content-api';

const GHOST_URL = 'http://blog.nafuna.tv';
const GHOST_KEY = '87f6096411ae42f96df2615620';
const GHOST_VERSION = 'v5';

export async function load() {
  const api = GhostContentAPI({
      url: GHOST_URL,
      key: GHOST_KEY,
      version: GHOST_VERSION})
  const postsJson = await api.posts.browse({limit: 5, include: 'tags,authors'});

  return {
    postsJson
  }
}
```
```svelte
<script lang="ts">
  // src/routes/+page.svelte
  import type { PageData } from './$types';

  export let data: PageData;
</script>

{#each data.postsJson as post}
  <h1> {post.title} </h1>
  <h3> {post.excerpt} </h3>
{/each}
```
Let me know if this helps!

**Reply by NafunaAfrica** [reply]
Hi there!!! 

Thank you so much for taking the time to respond. So I have tried your method but I still get an error 500. I copy pasted your first code into a server.ts file and updated the main page.svelte file and still throws up an error 500. I am attaching screenshot:
![image](https://user-images.githubusercontent.com/84924809/229325236-53704b9e-c5ce-425a-aa32-a7f79eeac7ac.png)

**Reply by NafunaAfrica** [reply]
![image](https://user-images.githubusercontent.com/84924809/229325306-bf697ecf-e2a7-4e6c-a78d-0c7bed46fe82.png)
Here is the current code

**Reply by theSalted** [reply]
Thank you for this!! Helped me a ton!

**Comment by NafunaAfrica** [comment]
Got the data in by adjusting the import of the SDK. The stuff is showing up. My next thing Is to understand how sveltekit would display the full page if I am to make each blog post a link so that is shows the bigger blog post: 
![image](https://user-images.githubusercontent.com/84924809/229325967-7af04958-e644-4c33-a0ac-458d233c9004.png)

**Reply by teemingc** [reply]
Awesome. I suggest reading up about slugs in the routing section https://kit.svelte.dev/docs/routing
The slug is dynamic and can be used to determine which post to load based on the slug value in the URL

**Reply by NafunaAfrica** [reply]
So I went through the documentation but still failing to get it to work because I think I am missing something on the svelte side:
![image](https://user-images.githubusercontent.com/84924809/229365713-5adab5fb-22fc-49c7-849a-63ff05f76276.png)
Please help! I think I am sooooo close!

**Reply by teemingc** [reply]
In the load function, you do not need to wrap the returned values with `prop`. Just return `{ post }` so you can unwrap it in the page as `data.post`

**Reply by NafunaAfrica** [reply]
![image](https://user-images.githubusercontent.com/84924809/229418085-4f09ebe7-d583-4ae9-a8bf-ffc31826fc33.png)
Like this?
this ius console:
![image](https://user-images.githubusercontent.com/84924809/229418228-28c5a5c4-f45c-4545-a5f3-0af7100410e4.png)

**Reply by NafunaAfrica** [reply]
I kow I am missing a small thing

**Reply by NafunaAfrica** [reply]
![image](https://user-images.githubusercontent.com/84924809/229418725-0ec2e205-4089-4274-b7fa-cdc21a70534a.png)
This is the chrome console log too

**Reply by teemingc** [reply]
I've created an example here for you
https://stackblitz.com/edit/sveltejs-kit-template-default-ib71fy?file=src/routes/post/[id]/+page.server.js
Let me know if it helps!

**Reply by NafunaAfrica** [reply]
Hello, I tried some of the statements on the side of the page.svelte and still get error 500.  So the error is coming from how the server.ts sends post. The error in vs code is: Property 'post' does not exist on type 'PageData'.ts(2339) when I hover over posi in the  $: ({ post } = data);

**Reply by teemingc** [reply]
I believe your issue lies with how the data is being returned in the load function in `+page.server.ts`.
Be sure that you're not wrapping the return value with `props`.
```diff
return {
-   props: { post }
+  { post }
}
```

**Reply by NafunaAfrica** [reply]
you have been an absolute Godsend! Its now working! the console is still throwing up errors but we are somewhere! Thank you so much! Now I will experiment with other feature later like pulling in only one post or filtering posts by tags! Awesome! This has me started!
